core/src/main/java/org/hibernate/ogm/jpa/impl/OgmStoredProcedureQuery.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.jpa.impl;
+
+import java.lang.invoke.MethodHandles;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+import javax.persistence.EntityManager;
+import javax.persistence.ParameterMode;
+import javax.persistence.StoredProcedureQuery;
+
+import org.hibernate.HibernateException;
+import org.hibernate.QueryParameterException;
+import org.hibernate.jpa.internal.StoredProcedureQueryImpl;
+import org.hibernate.jpa.spi.AbstractEntityManagerImpl;
+import org.hibernate.jpa.spi.HibernateEntityManagerImplementor;
+import org.hibernate.ogm.storedprocedure.impl.NoSQLProcedureCallImpl;
+import org.hibernate.ogm.storedprocedure.impl.NoSQLProcedureOutputImpl;
+import org.hibernate.ogm.util.impl.Log;
+import org.hibernate.ogm.util.impl.LoggerFactory;
+import org.hibernate.procedure.ParameterRegistration;
+import org.hibernate.procedure.internal.ProcedureCallMementoImpl;
+import org.hibernate.procedure.internal.ProcedureCallMementoImpl.ParameterMemento;
+
+/**
+ * @author Sergey Chernolyas &amp;sergey_chernolyas@gmail.com&amp;
+ */
+public class OgmStoredProcedureQuery extends StoredProcedureQueryImpl {
+
+	private static final Log log = LoggerFactory.make( MethodHandles.lookup() );
+	private Set<ParameterRegistration<?>> parameterRegistrations = new LinkedHashSet<>();
+	private NoSQLProcedureCallImpl procedureCall;
+	private HibernateEntityManagerImplementor entityManager;
+	private ProcedureCallMementoImpl procedureCallMemento;
+
+	public OgmStoredProcedureQuery(NoSQLProcedureCallImpl procedureCall, EntityManager entityManager) {
+		super( procedureCall, convert( entityManager ) );
+		this.procedureCall = procedureCall;
+		this.entityManager = convert( entityManager );
+	}
+
+	public OgmStoredProcedureQuery(NoSQLProcedureCallImpl procedureCall, EntityManager entityManager, ProcedureCallMementoImpl procedureCallMemento) {
+		this( procedureCall, entityManager );
+		this.procedureCallMemento = procedureCallMemento;
+		initProcedureCallByMemento();
+		procedureCall.setMemento( procedureCallMemento );
+	}
+
+	private void initProcedureCallByMemento() {
+		// register parameters
+		for ( ParameterMemento parameterMemento : procedureCallMemento.getParameterDeclarations() ) {
+
+			if ( procedureCall.getGridDialect().supportsNamedPosition() ) {
+				// @todo supports named position!
```

Comment:
This `if - then - else` does not seem right. Here you are preparing the parameters, shouldn't first check if the parameter is a named or positional parameter? Then, if it's a named paramter, you can check if the dialect support them and throw an exception.